e2e/tests/ui/features/@search/search.step.ts

```diff
@@ -0,0 +1,275 @@
+import { SearchPage } from "../../helpers/SearchPage";
+import { ToolbarTable } from "../../helpers/ToolbarTable";
+import { Tabs } from "../../helpers/Tabs";
+import { DetailsPage } from "../../helpers/DetailsPage";
+import { createBdd } from "playwright-bdd";
+import { expect } from "@playwright/test";
+
+export const { Given, When, Then } = createBdd();
+
+/**
+ * This function returns table identifier and column, which contains link to the details page
+ * @param type Catogory of the data to get the table identifier and column
+ */
+function getTableInfo(type: string): [string, string] {
+  switch (type) {
+    case "SBOMs":
+    case "SBOM":
+      return ["sbom-table", "Name"];
+    case "Advisories":
+    case "Advisory":
+      return ["advisory-table", "ID"];
+    case "Vulnerabilities":
+    case "CVE":
+      return ["Vulnerability table", "ID"];
+    case "Packages":
+    case "Package":
+      return ["Package table", "Name"];
+    default:
+      throw new Error(`Unknown type: ${type}`);
+  }
+}
+
+function getPaginationId(type: string): string {
+  switch (type) {
+    case "Vulnerabilities":
+      return "vulnerability-table-pagination-top";
+    case "Advisories":
+      return "advisory-table-pagination-top";
+    case "Packages":
+      return "package-table-pagination-top";
+    case "SBOMs":
+      return "sbom-table-pagination-top";
+    default:
+      throw new Error(`Unknown type: ${type}`);
+  }
+}
+
+function getColumns(type: string): string[] {
+  switch (type) {
+    case "Vulnerabilities":
+      return ["ID", "CVSS", "Date published"];
+    case "Advisories":
+      return ["ID", "Revision"];
+    case "Packages":
+      return ["Name", "Namespace", "Version"];
+    case "SBOMs":
+      return ["Name", "Created on"];
+    default:
+      throw new Error(`Unknown type: ${type}`);
+  }
+}
+
+Given("User is on the Search page", async ({ page }) => {
+  const searchPage = new SearchPage(page);
+  await searchPage.open();
+});
+
+Then(
+  "Download link should be available for the {string} list",
+  async ({ page }, type: string) => {
+    const table = new ToolbarTable(page, getTableInfo(type)[0]);
+    await table.verifyDownloadLink(type);
+  },
+);
+
+When(
+  "user starts typing a {string} in the search bar",
+  async ({ page }, searchText: string) => {
+    const searchPage = new SearchPage(page);
+    await searchPage.typeInSearchBox(searchText);
```

Comment:
I'm pretty sure methods for something like this already exist in the repo. Again, please, let's avoid redundant code and use what can be reused.